storage/mysql/tree_storage.go

```diff
@@ -104,6 +104,20 @@ func expandPlaceholderSQL(sql string, num int, first, rest string) string {
 	return strings.Replace(sql, placeholderSQL, parameters, 1)
 }
 
+// clearAllStmt clean up all sql.Stmt in cache
+func (m *mySQLTreeStorage) cleanAllStmt() {
+	m.statementMutex.Lock()
+	defer m.statementMutex.Unlock()
+
```

Comment:
Some kind of logging for this would be really useful. This should be a rare event in an otherwise healthy system:

`klog.Info("Clearing all prepared statements")`

Actually, for bonus points I think it's worth incrementing a counter whenever we clear statements, which should appear on monitoring dashboards and allow correlation with potential impacts such as increased latency or errors. Take a look at `queuedCounter` in this class as an example (though we won't need a `logID` for this as it's effectively global).

Reply:
fixed